File: 2.js
function fibonacci(limit, arr, prevPrevTerm, prevTerm) {
   var term = prevTerm + prevPrevTerm;
   if (term <= limit) 
      arr.push(term)
   else 
      return arr;
   return fibonacci(limit, arr, prevTerm, term);
}

function sumEvens(arr, result) {
   var result = result || 0;
   if (arr[0] % 2 == 0) 
      result += arr[0];
   if (arr.length == 1) 
      return result
   return sumEvens(arr.slice(1), result);
}

console.log(sumEvens(fibonacci(100, [1, 2], 1, 2)));

File: 2.scala
import scala.collection.mutable.ListBuffer

def fibonacci(limit: Int, arr: ListBuffer[Int], prevPrevTerm: Int, prevTerm: Int): List[Int] = {
	val term = prevTerm + prevPrevTerm;
	if (term <= limit) {
		arr += (term);
		fibonacci(limit, arr, prevTerm, term);
	}
	else arr.toList;
}

def sumEvens(nums: List[Int], result: Int = 0): Int = {
	nums.filter(a => a % 2 == 0).sum
}

val arr = new ListBuffer[Int]()
arr += 1
arr += 2
println(sumEvens(fibonacci(4000000, arr, 1, 2)))